students/BrianB/lesson03/mailroom1.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+from textwrap import dedent
+
+#----------------------------------------------
+# What: Lesson03: mailroom.py
+# Who: Brian Brumleve
+# Date: April 23, 2019
+# Program Description:
+#
+# To take something off the list in iPython:
+#  donor[1].pop(1)  Takes the 1 position
+#                       from donor 1
+#  donor[1].pop()  Leaving pop() blank, deletes
+#                       the value in the last
+#                       position
+#----------------------------------------------
+
+import sys
+import math
+from textwrap import dedent
+
+print("In the Mailroom")
+
+# convert lists to a dictionary - see Assignment 05, ToDo.py
+
+# A tuple of donor and listing of past donations
+# You want a tuple to keep a record of past donations. The original
+# tuple stays in as assigned memory location.
+donor_data = [("Guile", [1000, 589, 25000]),
+                ("Blanca", [23, 1, 17]),
+                ("Ken", [1001, 590, 25001]),
+                ("M. Bison", [15000, 8000, 1200000]),
+                ("Chun-Li", [61000000, 500000, 1200000]),
+                ]
+
+
+def thank_you():
+    while True:
+        name = input("Which Donor would you like to see?\n\
+    Please enter the full name only.\n\
+    >>> ")
+        if name == "list":
+            for donor in donor_data:
+                print(donor[0])
+            continue
+        for donor in donor_data:
```

Comment:
the logic is fine here, but too much embedded in the while loop.

Ideally, you call out to other functions for the actual work.